Support profiling for keda components (#1250)
kedacore/keda-docs: content/docs/2.13/troubleshooting.md

description = "How to address commonly encountered KEDA issues"
+++

> ## Troubleshoot keda errors using profiling

Reviewer: ```suggestion
## Troubleshoot KEDA errors using profiling
```

> in golang we have the possibility to profile specific actions in order to determine what causes an issue.
> For example, if our keda-operator pod is keeps getting OOM after a specific time, using profilig we can profile the heap and see what operatios taking all of this space.

Reviewer: ```suggestion
In Golang we have the possibility to profile specific actions in order to determine what causes an issue.
For example, if our `keda-operator` pod is keeps getting OOM after a specific time, using profilig we can profile the heap and see what operatios taking all of this space.
```

Golang support many profiling options like heap, cpu, goroutines and more... (for more info check this site https://pkg.go.dev/net/http/pprof).

> In Keda we provide the option to enable profiling on each component separately by enabling it using

Reviewer: ```suggestion
In KEDA we provide the option to enable profiling on each component separately by enabling it using
```

please use KEDA and not Keda :)

> In keda we provide the option to enable profiling on each component separately by enabling it using
> the helm chart and providing a port (if not enabled then it won't work)

Reviewer: ```suggestion
In Keda we provide the option to enable profiling on each component separately by enabling it using
the Helm chart and providing a port (if not enabled then it won't work).
```

```yaml
profiling:
  keda:
    enabled: false
    port: 8082

>   metricsApiServer:

Reviewer: please also incorporate changes from the charts here

Author: Can I just create it and bring reference to the keda docs?

Author: Fixed :)

    enabled: false
    port: 8083
  webhooks:
    enabled: false
    port: 8084
```

> If not using the helm chart then you can enabling the profiling on each on of the components by specifying
> this argument:

Reviewer: ```suggestion
If not using the Helm chart then you can enable the profiling on each on of components by specifying the
following argument in the respective container
```

```bash
--profiling-bind-address=":8082"
```

> and it will be expose on the port you specify.

Reviewer: ```suggestion
and it will be exposed on the port you specified.
```

> after enabling it you can port-forward or expose the service and use tool like go tool pprof in order to get profiling data.

Reviewer: ```suggestion
After enabling it you can port-forward or expose the service and use tool like go tool pprof in order to get profiling data.
```
